### ISSUE TITLE
Check that datetimes have the correct timezone

### ISSUE THREAD

**Issue by twiecki** [quantopian/alphalens#108]
We spent way too long debugging an odd exception:
```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-59-34a94da3a0f6> in <module>()
      2 
      3 alphalens.tears.create_factor_tear_sheet(factor=signal['signal'],
----> 4                                          prices=pricing, quantiles=2, periods=(1,))
      5                                          #show_groupby_plots=True)

/usr/local/lib/python2.7/dist-packages/alphalens/plotting.pyc in call_w_context(*args, **kwargs)
     41                 # sns.set_style("whitegrid")
     42                 sns.despine(left=True)
---> 43                 return func(*args, **kwargs)
     44         else:
     45             return func(*args, **kwargs)

/usr/local/lib/python2.7/dist-packages/alphalens/tears.pyc in create_factor_tear_sheet(factor, prices, groupby, show_groupby_plots, periods, quantiles, filter_zscore, groupby_labels, long_short, avgretplot, turnover_for_all_periods)
    117     mean_monthly_ic = perf.mean_information_coefficient(factor,
    118                                                         forward_returns,
--> 119                                                         by_time="M")
    120 
    121     factor_returns = perf.factor_returns(factor, forward_returns, long_short)

/usr/local/lib/python2.7/dist-packages/alphalens/performance.pyc in mean_information_coefficient(factor, forward_returns, group_adjust, by_time, by_group)
    133 
    134     else:
--> 135         ic = (ic.reset_index().set_index('date').groupby(grouper).mean())
    136 
    137     ic.columns = pd.Int64Index(ic.columns)

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in set_index(self, keys, drop, append, inplace, verify_integrity)
   2835                 names.append(None)
   2836             else:
-> 2837                 level = frame[col]._values
   2838                 names.append(col)
   2839                 if drop:

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in __getitem__(self, key)
   1995             return self._getitem_multilevel(key)
   1996         else:
-> 1997             return self._getitem_column(key)
   1998 
   1999     def _getitem_column(self, key):

/usr/local/lib/python2.7/dist-packages/pandas/core/frame.pyc in _getitem_column(self, key)
   2002         # get column
   2003         if self.columns.is_unique:
-> 2004             return self._get_item_cache(key)
   2005 
   2006         # duplicate columns & possible reduce dimensionality

/usr/local/lib/python2.7/dist-packages/pandas/core/generic.pyc in _get_item_cache(self, item)
   1348         res = cache.get(item)
   1349         if res is None:
-> 1350             values = self._data.get(item)
   1351             res = self._box_item_values(item, values)
   1352             cache[item] = res

/usr/local/lib/python2.7/dist-packages/pandas/core/internals.pyc in get(self, item, fastpath)
   3288 
   3289             if not isnull(item):
-> 3290                 loc = self.items.get_loc(item)
   3291             else:
   3292                 indexer = np.arange(len(self.items))[isnull(self.items)]

/usr/local/lib/python2.7/dist-packages/pandas/indexes/base.pyc in get_loc(self, key, method, tolerance)
   1945                 return self._engine.get_loc(key)
   1946             except KeyError:
-> 1947                 return self._engine.get_loc(self._maybe_cast_indexer(key))
   1948 
   1949         indexer = self.get_indexer([key], method=method, tolerance=tolerance)

pandas/index.pyx in pandas.index.IndexEngine.get_loc (pandas/index.c:4154)()

pandas/index.pyx in pandas.index.IndexEngine.get_loc (pandas/index.c:4018)()

pandas/hashtable.pyx in pandas.hashtable.PyObjectHashTable.get_item (pandas/hashtable.c:12368)()

pandas/hashtable.pyx in pandas.hashtable.PyObjectHashTable.get_item (pandas/hashtable.c:12322)()

KeyError: 'date'
```

Just to find out that the reason was that the factor index we passed was not UTC localized. Since we only support daily data the timezone does not matter in the first place so we should just cast it to be UTC.

**Comment by jameschristopher** [comment]
Agreed, I've also spent time debugging this confusing exception. I'll add to the 0.1 release milestone.

**Comment by jameschristopher** [comment]
fixed #130
